src/SFML/Window/Win32/JoystickImpl.cpp

```diff
@@ -302,9 +496,19 @@ void JoystickImpl::close()
         closeDInput();
 }
 
+
 ////////////////////////////////////////////////////////////
 JoystickCaps JoystickImpl::getCapabilities() const
 {
+    if (m_useXInput)
+    {
+        // XInput has 10 Buttons (since we exclude the DPad) and all 8 Axes
+        JoystickCaps caps{0};
+        caps.buttonCount = 10;
+        caps.axes.fill(true);
+        return caps;
+    }
+
     if (directInput)
         return getCapabilitiesDInput();
```

Comment:
Maybe `DInput` should be the fallback, and the entire `joyXXX` legacy api can be dropped?
@ChrisThrasher what do you think about that?